docs/architecture/libindy_layers.md

```diff
@@ -0,0 +1,65 @@
+There are 3 layers in libindy code organization:
+* API layer
+* Commands Executors layer
+* Services layer
+
+Each of them contains multiply modules devoted to particular part of the flow, like "wallet", "pool", "crypto", etc. 
+
+## Layers purposes
```

Comment:
1) It would be great to emphasize the main thing about the Commands and Service layers: that Commands layer splits complex operation to atomic ones and delegate execution of atomic operation to Services where Services doesn't communicate with each other.
2) It would be great to emphasize the benefits of such encapsulation
3) It would be great to describe how secure enclave works in libindy

Reply:
1. fixed
2. not sure, that arch overview is a place to for that. We don't have any specific benefits in libindy, just well-known advantages like testability, easier code support, etc
3. already present in design folder